crates/sb_ai/utils.rs

```diff
@@ -0,0 +1,151 @@
+use std::{hash::Hasher, path::PathBuf};
+
+use anyhow::{bail, Context};
+use deno_core::error::AnyError;
+use futures::{io::AllowStdIo, StreamExt};
+use reqwest::Url;
+use tokio::io::AsyncWriteExt;
+use tokio_util::compat::FuturesAsyncWriteCompatExt;
+use tracing::{debug, error, info, info_span, instrument, trace, Instrument};
+use xxhash_rust::xxh3::Xxh3;
+
+#[instrument(fields(%kind, url = %url))]
+pub async fn fetch_and_cache_from_url(
+    kind: &'static str,
+    url: Url,
+    cache_id: Option<String>,
+) -> Result<PathBuf, AnyError> {
+    let cache_id = cache_id.unwrap_or(fxhash::hash(url.as_str()).to_string());
+    let download_dir = ort::sys::internal::dirs::cache_dir()
```

Comment:
is it possible to configure this when starting the edge-runtime?

Reply:
resolved

Reply:
`SB_AI_CACHE_DIR`
https://github.com/supabase/edge-runtime/blob/9e0a5633b83066b1624e496d5ada99193802b5be/crates/sb_ai/utils.rs#L19